### PR TITLE
[api docs] NestedParamsEncoder

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR adds a bit of API documentation to an undocumented class.

Questions for reviewer:

- [x] Perhaps you know a narrative comment for the description of this class?

### CODE REVIEW

File: lib/faraday/parameters.rb [hunk 1]
```diff
@@ -63,6 +68,11 @@ def self.encode(params)
       return buffer.chop
     end
 
+    # @param query [nil, String]
+    #
+    # @raise [TypeError] if the nesting is incorrect
+    #
+    # @return [Array<Array, String>] the decoded params
```
Comment (iMacTia) [inline lib/faraday/parameters.rb]: I was looking for a convention on tags order, but I couldn't find anything.
At least I think we should try to be consistent. I prefer the order of the method above so can we align this one to it (`@param`, `@return`, `@raise`)?

Comment (olleolleolle) [reply lib/faraday/parameters.rb]: 👍

**Comment by olleolleolle** [comment]
@iMacTia Thanks for the input and the class-level comment.

**Comment by olleolleolle** [comment]
@iMacTia And I would like to thank the participants in the workshop at Malmo.rb where we did the investigation and markup work!

**Comment by iMacTia** [comment]
Oh wow didn't know that was a team effort! Well done everyone 🥇 !

**Pull request merged**